bootloader/lkenv/lkenv.go

```diff
@@ -81,142 +159,117 @@ func copyString(b []byte, s string) {
 	}
 }
 
-func NewEnv(path string) *Env {
-	return &Env{
+func NewEnv(path string, version Version) *Env {
+	e := &Env{
 		path:    path,
 		pathbak: path + "bak",
-		env: SnapBootSelect_v1{
+		version: version,
+	}
+
+	switch version {
+	case V1:
+		// TODO: move the initial assignment of members to setup() ?
```

Comment:
I had a slightly different design in mind when I proposed setup. Doesn't matter,  here it seems more a case of having newFoo functions

Reply:
I added new* functions here